### PR TITLE
Upgrade Arrow to 37

### PR DESCRIPTION
**Pull request by changhiskhan**
Can't go to 38 yet even though it's out. Datafusion is on 37 latest.

We need 37 for Will's fix for handling schema metadata.

### CODE REVIEW

File: rust/src/arrow/linalg.rs [hunk 1]
```diff
@@ -254,7 +251,7 @@ impl MatrixView {
         let mut builder = Float32Builder::with_capacity(n * dim);
         for idx in chosen.iter() {
             let s = self.data.slice(idx * dim, dim);
-            builder.append_slice(as_primitive_array::<Float32Type>(s.as_ref()).values());
+            builder.append_slice(s.values());
```
Comment (eddyxu) [inline rust/src/arrow/linalg.rs:254]: Slice is strong typed now?

File: rust/src/arrow/linalg.rs [hunk 2]
```diff
@@ -447,13 +444,18 @@ mod tests {
...
             0.10910681,
         ];
-        assert_relative_eq!(u.data().values(), expected_u.as_slice(), epsilon = 0.0001,);
-
-        assert_relative_eq!(
-            sigma.values(),
-            vec![27.46873242, 22.64318501, 8.55838823, 5.9857232, 2.01489966].as_slice(),
-            epsilon = 0.0001,
-        );
+        u.data()
```
Comment (eddyxu) [inline rust/src/arrow/linalg.rs:447]: why can we just use slice comparision here?

**Review by eddyxu — APPROVED** [review]
Seems fine to me, other than a few concept changes , i assume that is what is done in Arrow

File: rust/src/arrow/linalg.rs (hunk 1 above)
Comment (changhiskhan) [reply rust/src/arrow/linalg.rs:254]: yeah, must be part of the effort towards arrow-rs arrow2 merging?

File: rust/src/arrow/linalg.rs (hunk 2 above)
Comment (changhiskhan) [reply rust/src/arrow/linalg.rs:447]: complains about ScalarBuffe<f32> not having AbsDiffEq

**Pull request merged**